matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/util/MatrixItem.kt

```diff
@@ -35,7 +35,20 @@ sealed class MatrixItem(
     data class UserItem(override val id: String,
                         override val displayName: String? = null,
                         override val avatarUrl: String? = null) :
-        MatrixItem(id, displayName?.removeSuffix(ircPattern), avatarUrl) {
+            MatrixItem(id, displayName?.removeSuffix(IRC_PATTERN), avatarUrl) {
+        init {
+            if (BuildConfig.DEBUG) checkId()
+        }
+
+        override fun updateAvatar(newAvatar: String?) = copy(avatarUrl = newAvatar)
+    }
+
+    // TODO is it correct to represent it by a Matrix Item ? => to confirm
```

Comment:
I was asking myself if the "@room" tag can be represented as a `MatrixItem` since in messages, it is just raw text and it is not converted to any matrix item link like others. I choose to represent it as `MatrixItem` since it was the easiest way to implement it.

Reply:
This approach is fine to me, thanks!